lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +27,31 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
-        response = compute_client.list_instances(compartment_id, options)
+        servers = {}
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        instances = compute_client.list_instances(compartment_id, options)
+
+        # build a hash of hashs, keyed off of server_id
+        vnics = compute_client.list_vnic_attachments(compartment_id, {})
+        vnics.data.each do |vnic|
+          servers[vnic.instance_id] = {}
+          servers[vnic.instance_id]['instance_id'] = vnic.instance_id
+          servers[vnic.instance_id]['vnic_id'] = vnic.vnic_id
+        end
+
+        servers.each do |server, data|
+          vnic = network_client.get_vnic(data['vnic_id'], {})
+          servers[server]['private_ip'] = vnic.data.private_ip
+          servers[server]['public_ip'] = vnic.data.public_ip
+        end
+
+        display_list(instances,
+          ['Display Name', 'State', 'Public IP', 'Private IP', 'ID']) do |item|
+          [item.display_name,
+           item.lifecycle_state,
+           servers[item.id]['public_ip'],
```

Comment:
Note that this will show public and private IP when using the default summary format, but will not show it in other formats. I think that's fine - for the other formats it makes sense that they would show the raw instance data which does not include this. However, we might consider checking the format in advance and not bothering to get this data if the view is not summary.